src/frontends/paddle/src/internal/pass/transform_fakequantize.cpp

```diff
@@ -0,0 +1,110 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "internal/pass/transform_fakequantize.hpp"
+
+#include <ngraph/ngraph.hpp>
+#include <ngraph/op/util/op_types.hpp>
+#include <ngraph/pattern/matcher.hpp>
+#include <ngraph/pattern/op/or.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/variant.hpp>
+
+#include "default_opset.hpp"
+#include "openvino/pass/pattern/op/label.hpp"
+
+using namespace ov::frontend::paddle::op::default_opset;
+using namespace ov;
+using namespace ov::pass;
+using namespace ov::frontend::paddle::op;
+
+/*
+                                                                       x
+                                                                   |  |  |  |
```

Comment:
I have a fundamental question here -
Why not directly map the framework "quantize_linear" and "dequantize_linear" to frontend internal ops and further transform them to FakeQuantize?

Reply:
I rechecked Sergey's comments, and I understand now that since there is a MOC transformation already, to follow onnx might be a better solution.

Reply:
> I rechecked Sergey's comments, and I understand now that since there is a MOC transformation already, to follow onnx might be a better solution.

- Also agree with Sergey's comments. But when translating the PaddleSlim quantize_linear and dequantize_linear op, more datatype and shape conversion are imported, which is not same as ONNX. As a result, the graph translated cant hit the LPT pass and the performance is low. That's why I import a custom pass in the PDPD FE.
- It is to keep it moving forward. If reuse LTP pass in this milestone, some common pass may need modification with some risk. So this implementation in PDPD FE is clean relatively and doesn't break the common pass. 
- But it is not STOP. I will still optimize it and possible LPT pass in the next wave. The last target is to reuse the LPT pass as ONNX does instead of importing the custom pass.

Reply:
> > I rechecked Sergey's comments, and I understand now that since there is a MOC transformation already, to follow onnx might be a better solution.
> 
> * Also agree with Sergey's comments. But when translating the PaddleSlim quantize_linear and dequantize_linear op, **more datatype and shape transpose are imported to meet the quantization algorithm, which is not same as ONNX.** As a result, the graph translated cant hit the LPT pass and the performance is low. That's why I import a custom pass in the PDPD FE.
> * It is to keep it moving forward. If reus**e LTP pass in this milestone, some common pass may need modification with some risk.** So this implementation in PDPD FE is clean relatively and doesn't break the common pass.
> * But it is not STOP. I will still optimize it and possible LPT pass in the next wave. The last target is to reuse the LPT pass as ONNX does instead of importing the custom pass.


Probably you can elaborate this discrepancy and challenge in our weekly.

Reply:
> > > I rechecked Sergey's comments, and I understand now that since there is a MOC transformation already, to follow onnx might be a better solution.
> > 
> > 
> > 
> > * Also agree with Sergey's comments. But when translating the PaddleSlim quantize_linear and dequantize_linear op, **more datatype and shape transpose are imported to meet the quantization algorithm, which is not same as ONNX.** As a result, the graph translated cant hit the LPT pass and the performance is low. That's why I import a custom pass in the PDPD FE.
> > * It is to keep it moving forward. If reus**e LTP pass in this milestone, some common pass may need modification with some risk.** So this implementation in PDPD FE is clean relatively and doesn't break the common pass.
> > * But it is not STOP. I will still optimize it and possible LPT pass in the next wave. The last target is to reuse the LPT pass as ONNX does instead of importing the custom pass.
> 
> Probably you can elaborate this discrepancy and challenge in our weekly.

Yes. There is a talk about this. I will add you in e-mail.

Reply:
Probably we can get this solution merged for this release considering it is asked by customers with high priority. @slyalin 
 How do you think?

@xczhai Would you please create a ticket to make sure to investigate and follow ONNX path in next milestone?

Reply:
> Probably we can get this solution merged for this release considering it is asked by customers with high priority. @slyalin How do you think?
> 
> @xczhai Would you please create a ticket to make sure to investigate and follow ONNX path in next milestone?

@ceciliapeng2011 OK. 104434 has been created for track. thx.